File: adobe-edge/src/api/details/AdobeChapterDetails.ts
/**
 * Chapter details information.
 *
 * https://github.com/adobe/xdm/blob/master/docs/reference/datatypes/chapterdetails.schema.md
 */
export interface AdobeChapterDetails {
  // The ID of the ad break.
  ID?: string;

  // The friendly name of the Ad Break.
  friendlyName?: string;

  // The position (index, integer) of the chapter inside the content.
  index: number;

  // Chapter is completed.
  isCompleted?: boolean;

  // Chapter is started.
  isStarted?: boolean;

  // The length of the chapter, in seconds.
  length: number;

  // The offset of the chapter inside the content (in seconds) from the start.
  offset: number;

  // The time spent on the chapter, in seconds.
  timePlayed?: number;
}

File: adobe-edge/android/src/main/java/com/theoplayer/reactnative/adobe/edge/api/AdobeChapterDetails.kt
package com.theoplayer.reactnative.adobe.edge.api

/**
 * Chapter details information.
 *
 * @see <a href="https://github.com/adobe/xdm/blob/master/docs/reference/datatypes/chapterdetails.schema.md">Adobe XDM ChapterDetails Schema</a>
 */
data class AdobeChapterDetails(
  // The ID of the chapter.
  val id: String? = null,
  // The friendly name of the chapter.
  val friendlyName: String? = null,
  // The position (index, integer) of the chapter inside the content.
  val index: Int,
  // Chapter is completed.
  val isCompleted: Boolean? = null,
  // Chapter is started.
  val isStarted: Boolean? = null,
  // The length of the chapter, in seconds.
  val length: Int,
  // The offset of the chapter inside the content (in seconds) from the start.
  val offset: Int,
  // The time spent on the chapter, in seconds.
  val timePlayed: Int? = null
)
